agent/util-scripts/pbench-tool-meister-start

```diff
@@ -272,44 +384,102 @@ def main(argv):
     sh.setFormatter(shf)
     logger.addHandler(sh)
 
-    try:
-        group = argv[1]
-    except IndexError:
-        group = "default"
+    # +
+    # Step 1 - Load the tool group data for the requested tool group
+    # -
 
-    # 1. Load the tool group data given the tool group argument
+    # Verify all the command line arguments
+    group = cli_params.tool_group
     try:
+        # Load the tool group data
         tool_group = ToolGroup(group)
-    except Exception:
-        logger.exception("failed to load tool group data")
+    except BadToolGroup as exc:
+        logger.error(str(exc))
         return 1
+    except Exception:
+        logger.exception("failed to load tool group data for '%s'", group)
+        return ReturnCode.BADTOOLGROUP
+    else:
+        if not tool_group.hostnames:
+            # If a tool group has no tools registered, then there will be no
+            # host names on which to start Tool Meisters.
```

Comment:
Consider adding a warning or at least some message to say no tools are registered on hosts, so no Tool Meisters will be started.

Reply:
Tracked in https://github.com/distributed-system-analysis/pbench/issues/2164.